### ISSUE TITLE
Feature request: svg.toDataUrl

### ISSUE THREAD

**Issue by msageryd** [expo/expo#4318]
## 🐛 toDataURL() function is missing in universal module for SVG.

@evanbacon, Here is the issue we talked about at react-europe. I'm not sure of how much information you need. Please tell if there is anything else I can do. I'll help testing the function, for sure. But unfortunately I'm not able to help with the implementation.

### Use case

It would be really cool to be able to use toDataURL in react-native-web. To me the use case is to have a server being able to render some complex SVG that is currently rendered only in the app. Without toDataURL I would need to maintain two code bases for my SVG graphics, one for RN and one for ordinary web SVG.

### Implementation

There are probably some things that needs to be considered:

- toDataURL has some timing issues in react-native-svg. To ensure that everything is rendered, toDataURL needs to be called from onLayout in the parent View.

- Since the timing problem is worked around via `onLayout`, we need to ensure that toDataURL is called only once. This might be more of a documentation issue, i.e. show the users how to use toDataURL.

- In react-native-svg, the output size of the bitmap is multiplied by the pixel density of the device, i.e. on an iPhoneX the below code will render a 3000x3000 file. This is an almost solved issue, but some quirks are still there (see linked issue below).

- To make an actual PNG out of the base64 you will need to strip some stuff from the beginning of the data as well as strip all linefeeds. I don't know when or why the linefeeds are inserted into the data, but I'm guessing it has to do with buffer concatenation or similar, because they only appear in very large files. This might also be a an issue for the docs only, but I suspect that PNG output is a common use for toDataURL.

```
render = () => {
  return (
    <View onLayout={this.onSvgLayout}>
      <Svg
        viewBox={'0 0 1000 1000'}
        ref={ref => {
          this.svg = ref;
        }}
      >
        //your SVG graphics..
      </Svg>
    </View>
  );
};

onSvgLayout = () => {
  //Ensure that only one call to onSvgLayout is doing work
  if (this.state.base64) return null;

  this.svg.toDataURL(
    base64 => {
      this.setState({base64});

      //Strip some initial stuff to make this a working PNG file
      let fileData = base64.replace(/^data:image\/\w+;base64,/, '');

      //Also strip line-breaks that seems to be inserted in very large files
      fileData = fileData.replace(/\r?\n|\r/g, '');

      RNFetchBlob.fs
        .createFile('filename.png', fileData, 'base64')
        .then(() => console.log('base64 file written'))
        .catch(e => console.log(e));
    },
    {width: 1000, height: 1000}
  );
};

```
See: https://github.com/react-native-community/react-native-svg/issues/855

**Comment by EvanBacon** [comment]
@msand would you be interested in taking a pass at this?

**Comment by FiberJW** [comment]
We currently import Svg directly from https://github.com/react-native-community/react-native-svg, so if you could create an issue there, you're more likely to get a reply from someone more involved in the project. If you have any further related problems, please open another issue. Thanks :)

**Comment by msand** [comment]
Maybe this could be used? https://github.com/sampumon/SVG.toDataURL/blob/master/index.js

**Comment by msageryd** [comment]
As of now I'm actually maintaining separate code bases for my RN SVG graphics and my standard html SVG graphics. The geometry calculations is shared code, though. The reason is that I need to use the html in jsreport for rendering pdf reports. I'm using a simple node/Express server to render the SVGs.

I no longer have use for toDataUrl, but I do think it would be a usable function. As the issue initiator I thought I'd chime in with this new information.

If you don't think this is a good function to have in Expo in general, please go ahead and close this issue.

**Comment by msand** [comment]
Well, at least the most basic logic for it from that lib is not too many bytes, can probably add it to react-native-svg, probably won't include canvg for older browsers etc. if people need more backward compatibility they can simply use that library in addition.
